File: julia/matching-brackets/matching-brackets.jl
function matching_brackets(input)
    cleaned = join([c for c in input if c in "(){}[]"])
    while true
        old_len = length(cleaned)
        cleaned = replace(cleaned, "()"=>"")
        cleaned = replace(cleaned, "[]"=>"")
        cleaned = replace(cleaned, "{}"=>"")
        if length(cleaned) == old_len
            break
        end
    end
    cleaned == ""
end

File: fsharp/matching-brackets/MatchingBrackets.fs
module MatchingBrackets

let isPaired input =
    let brackets = "(){}[]" |> Seq.toArray
    let cleaned =
        String.collect (fun c -> if Seq.exists((=)c) brackets then string c else "") input
        
    let rec remove_pairs text =
        let start_len = String.length text
        let modified = text.Replace("()", "").Replace("{}", "").Replace("[]", "")
        if String.length modified = start_len then modified else remove_pairs modified
        
    String.length (remove_pairs cleaned) = 0
    